clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1484,6 +1484,8 @@ public void testRebalanceWithMetadataChange() {
                 Utils.mkMap(Utils.mkEntry(topic1, 1), Utils.mkEntry(topic2, 1))));
         client.respond(joinGroupFollowerResponse(1, consumerId, "leader", Errors.NOT_COORDINATOR));
         client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.poll(time.timer(0)); // failing joinGroup request will require re-poll in order to retry
```

Comment:
It's not very clear to me why here and line 3403 below we need additional polls since the test scenarios seems irrelevant to error cases?

Reply:
good point... let me update the tests there. thanks

Reply:
The NOT_COORDINATOR error originally should trigger retries; however, in the new code, it would exit due to an expired timer. Another way to do it is using poll(time.timer(1))

Reply:
Ack, that makes sense.